### DISCUSSION TITLE
how to update existing files when creating a project

### DISCUSSION THREAD

**Discussion by mgaitan** [copier-org/copier#670] [Q&A] [answered]
Hi, I'm new with copier and I'm very glad to have found such a good tool. 

I'm trying to use it to generate new subpackages inside a project that already exists, while also update some existing files. 

For instance,  my template currently is like this

```bash
📁 my_copier_template 
├── 📄 copier.yml
├── 📁 existing_package/
│    └── 📁  {{ subpackage_name }}
│            └── 📄 {{ module_name }}.py
```
and I got the subpackage in my existing package

```bash
├── 📁 existing_package/
│    └── 📄 existing_module.py
│    └── 📁  existing_subpackage1
│    └── 📁  <subpackage_name>
│            └── 📄 <module_name>.py
```

But now I want to update also `existing_module.py`, for instance to append something after a comment that matches a regex pattern. 

Is that possible?

**Comment by yajo** [comment, marked as answer]
Sounds like you want some custom behavior. Try using tasks.
https://copier.readthedocs.io/en/stable/configuring/#tasks

**Comment by mgaitan** [comment]
I've implemented a script that works for a very basic use case: when I need to append a new content to my 
- Create a file `after-edit.py` in the root of the template (beside `copier.yaml`) 

```python 
from pathlib import Path
from copier.tools import Style, printf

printf("Running after edit script")
for to_append in Path('.').glob('**/*.append'):
    text_to_append = to_append.read_text()
    target = Path(str(to_append).rstrip(".append"))
    printf("Appended into", target, style=Style.WARNING)

    target.write_text(target.read_text() + text_to_append)
    to_append.unlink()
Path("after-edit.py").unlink()
```
- Add a file `existing_package/existing_module.py.append.jinja` to create the content to append
- Include the task in the config file

```yaml
 _tasks:
  - ["{{ _copier_python }}", after-edit.py] 
```

PS: I'm wondering if I could run the script without copying it to the target folder (and then self-deleting it from the script's last line).

**Reply by yajo** [reply]
About your last question, see how it's done for migrations here: https://github.com/Tecnativa/doodba-copier-template/blob/11e31c188084449f99dca38de739a4b25fe0c9db/copier.yml#L47
